test/MusicStore.Test/CartSummaryComponentTest.cs

```diff
@@ -0,0 +1,109 @@
+﻿using System;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNet.Http;
+using Microsoft.AspNet.Mvc;
+using Microsoft.AspNet.Mvc.ModelBinding;
+using Microsoft.AspNet.Session;
+using Microsoft.Framework.Caching.Distributed;
+using Microsoft.Framework.Caching.Memory;
+using Microsoft.Framework.DependencyInjection;
+using Microsoft.Framework.Logging.Testing;
+using MusicStore.Models;
+using Xunit;
+
+namespace MusicStore.Components
+{
+    public class CartSummaryComponentTest
+    {
+        private readonly IServiceProvider _serviceProvider;
+
+        public CartSummaryComponentTest()
+        {
+            var services = new ServiceCollection();
+
+            services.AddEntityFramework()
+                      .AddInMemoryStore()
+                      .AddDbContext<MusicStoreContext>();
+
+            _serviceProvider = services.BuildServiceProvider();
+        }
+
+        [Fact]
+        public async Task CartSummaryComponent_Returns_CartedItems()
+        {
+            // Arrange
+            var viewContext = new ViewContext()
+            {
+                ViewData = new ViewDataDictionary(
+                    new EmptyModelMetadataProvider(),
+                    new ModelStateDictionary()),
+                HttpContext = new DefaultHttpContext()
+            };
+
+            // Session initialization
+            var cartId = "CartId_A";
+            var sessionFeature = new SessionFeature()
```

Comment:
@Eilon 
Did you take a look at this?? I think MusicStore using the session feature. 
https://github.com/aspnet/MusicStore/blob/dev/src/MusicStore/Models/ShoppingCart.cs#L150-L164